.github/workflows/git-commit-checks.py

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env python3
+
```

Comment:
For the next person, can you include what this script does, its inputs, and its outputs?  Maybe also what output goes where?  I'm having trouble figuring out what the output will look like in the GitHub ecosystem.

How does one test this code without making changes in production?

In particular, I'm sure there's a good reason, but I can't figure out why the cherry pick logic needs to live in the yaml configuration file instead of with the rest of the business logic in this file.  It's a very weird separation of concerns.

Reply:
1. Docs: yes, I should (will) definitely do that.
2. Testing: I've been developing/testing this in my own personal repository.  This seems to be an oddity in Github: the *first* github action you create (or potentially any *new* github action? I didn't chase down the difference) doesn't seem to get run until *after* it is present on the base branch.  After that, any future PR you make that changes the github action code -- the YAML or the code that the YAML invokes -- will be invoked by the PR.  Meaning: the *first* PR creating a github action is weird.  After that, it behaves like you expect (i.e., you can test/experiment on the PR before merging to PR to production).
3. Cherry pick logic: fair enough.  The YAML has a built-in github token, making it trivial to issue github API calls.  Specifically: the code, as it currently stands, has all github API stuff in the YAML; the Python does only local git operations.  I can change that to figure out how to move the github API stuff into the Python.

Reply:
>     3. Cherry pick logic: fair enough.  The YAML has a built-in github token, making it trivial to issue github API calls.  Specifically: the code, as it currently stands, has all github API stuff in the YAML; the Python does only local git operations.  I can change that to figure out how to move the github API stuff into the Python.

That makes sense.  If you add that in a comment, I'm happy with leaving the code the way it is.

Reply:
Nah, it was actually straightforward enough to pull the github API stuff into the Python.  It was a good suggestion; it ended up making the overall code cleaner.